src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/FoldingRanges/RemoteFoldingRangeService.Factory.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Razor.Remote;
+
+namespace Microsoft.CodeAnalysis.Remote.Razor;
+
+internal sealed partial class RemoteFoldingRangeService
+{
+    internal sealed class Factory : FactoryBase<IRemoteFoldingRangeService>
```

Comment:
As mentioned offline, I think it would be perfectly reasonable to simplify things and put the factories in the service classes. Simplifies copy-and-paste programming at least.